tools/libSwiftSyntaxParser/libSwiftSyntaxParser.cpp

```diff
@@ -196,9 +273,14 @@ swiftparse_client_node_t SynParser::parse(const char *source) {
   ParserUnit PU(SM, SourceFileKind::Library, bufID, langOpts,
                 "syntax_parse_module", std::move(parseActions),
                 /*SyntaxCache=*/nullptr);
+  std::unique_ptr<SynParserDiagConsumer> pConsumer;
```

Comment:
This does not seem to be used outside the `if` block, you can declare it inside the block.

Reply:
We cannot put it inside the if block because then the consumer is getting deleted before `parse()`.

Reply:
Ah, I see. 👍